source/common/event/libevent_scheduler.h

```diff
@@ -47,6 +47,13 @@ class LibeventScheduler : public Scheduler {
    */
   void initializeStats(DispatcherStats* stats);
 
+  /**
+   * Registers callback to be called in the event loop prior to polling for
+   * events. Overwrites |callback| from previous calls. Calling with null
```

Comment:
Does evwatch_prepare_new() override previous callbacks? From my reading of the definition, it doesn't override but append another watcher. But I might miss something critical.

Reply:
evwatch_prepare_new() does not, but it's only called once in LibeventScheduler constructor.  Each call of the registerOnPrepareCallback() method overrides the previous value of LibeventScheduler::callback_.

There is no way to delete a callback added by evwatch_prepare_new(), so this is necessary in order to be able to unregister callbacks when EnvoyQuicClock() is destroyed.